docs/user/ppl/cmd/join.rst

```diff
@@ -11,27 +11,35 @@ join
 
 Description
 ===========
-| (Experimental)
-| (From 3.0.0)
 | Using ``join`` command to combines two datasets together. The left side could be an index or results from a piped commands, the right side could be either an index or a subsearch.
 
 Version
 =======
 3.0.0
 
-Syntax
-======
-[joinType] join [leftAlias] [rightAlias] on <joinCriteria> <right-dataset>
+Basic syntax in 3.0.0
+=====================
+| [joinType] join [leftAlias] [rightAlias] (on | where) <joinCriteria> <right-dataset>
 
-* joinType: optional. The type of join to perform. The default is ``INNER`` if not specified. Other option is ``LEFT [OUTER]``, ``RIGHT [OUTER]``, ``FULL [OUTER]``, ``CROSS``, ``[LEFT] SEMI``, ``[LEFT] ANTI``.
+* joinType: optional. The type of join to perform. The default is ``inner`` if not specified. Other option is ``left``, ``semi``, ``anti`` and performance sensitive types ``right``, ``full`` and ``cross``.
 * leftAlias: optional. The subsearch alias to use with the left join side, to avoid ambiguous naming. Fixed pattern: ``left = <leftAlias>``
 * rightAlias: optional. The subsearch alias to use with the right join side, to avoid ambiguous naming. Fixed pattern: ``right = <rightAlias>``
-* joinCriteria: mandatory. It could be any comparison expression.
-* right-dataset: mandatory. Right dataset could be either an index or a subsearch with/without alias.
+* joinCriteria: mandatory. It could be any comparison expression. Must follow with ``on`` (since 3.0.0) or ``where`` (since 3.3.0) keyword.
```

Comment:
Cross join also requires join condition?

Reply:
yes. it requires the join condition such as `on 1=1`. BTW, the cross join disabled by default from 3.3.0